core/src/main/scala/kafka/log/Log.scala

```diff
@@ -1896,6 +1994,11 @@ class Log(@volatile private var _dir: File,
             val deletable = logSegments.filter(segment => segment.baseOffset > targetOffset)
             removeAndDeleteSegments(deletable, asyncDelete = true, LogTruncation)
             activeSegment.truncateTo(targetOffset)
+
+            // TODO: @kamalcph check the logic again
+            this.localLogStartOffset = math.min(targetOffset, this.localLogStartOffset)
+            updateLogStartOffset(math.min(this.localLogStartOffset, this.logStartOffset))
```

Comment:
These two lines do not align with the `truncateFullyAndStartAt` case above.
The `truncateFullyAndStartAt` calls `completeTruncatition` to update the logStartOffset, and there is already a completeTruncation below on line 2004. So are these two lines still required?

Reply:
Discussed with Uber, we can remove the redundant check.